src/hyperlight_common/src/flatbuffer_wrappers/guest_trace_data.rs

```diff
@@ -0,0 +1,672 @@
+/*
+Copyright 2025  The Hyperlight Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//! Guest trace data structures and (de)serialization logic.
+//! This module defines the data structures used for tracing spans and events
+//! within a guest environment, along with the logic for serializing and
+//! deserializing these structures using FlatBuffers.
+//!
+//! Schema definitions can be found in `src/schema/guest_trace_data.fbs`.
+
+use alloc::string::{String, ToString};
+use alloc::vec::Vec;
+
+use anyhow::{Error, Result, anyhow};
+use flatbuffers::size_prefixed_root;
+
+use crate::flatbuffers::hyperlight::generated::{
+    CloseSpanType as FbCloseSpanType, CloseSpanTypeArgs as FbCloseSpanTypeArgs,
+    GuestEventEnvelopeType as FbGuestEventEnvelopeType,
+    GuestEventEnvelopeTypeArgs as FbGuestEventEnvelopeTypeArgs, GuestEventType as FbGuestEventType,
+    GuestTraceDataType as FbGuestTraceDataType, GuestTraceDataTypeArgs as FbGuestTraceDataTypeArgs,
+    KeyValue as FbKeyValue, KeyValueArgs as FbKeyValueArgs, LogEventType as FbLogEventType,
+    LogEventTypeArgs as FbLogEventTypeArgs, OpenSpanType as FbOpenSpanType,
+    OpenSpanTypeArgs as FbOpenSpanTypeArgs,
+};
+
+/// Key-Value pair structure used in tracing spans/events
+#[derive(Debug, Clone)]
+pub struct KeyValue {
+    /// Key of the key-value pair
+    pub key: String,
+    /// Value of the key-value pair
+    pub value: String,
+}
+
+impl TryFrom<FbKeyValue<'_>> for KeyValue {
+    type Error = Error;
+
+    fn try_from(value: FbKeyValue<'_>) -> Result<Self, Self::Error> {
+        let key = value.key().to_string();
+        let value = value.value().to_string();
+
+        Ok(KeyValue { key, value })
+    }
+}
```

Comment:
nit: if this is is infallible you can implement `From` rather than `TryFrom`

Reply:
Fixed. I changed to implement `From`